R/MsExperiment.R

```diff
@@ -19,62 +19,73 @@ setMethod("saveMsObject",
                          recursive = TRUE,
                          showWarnings = FALSE)
               ## sample data
-              write.table(as.data.frame(object@sampleData), sep = "\t",
+              sdata <- object@sampleData
+              if (!length(sdata)) # initialize with empty data frame
```

Comment:
that's a workaround to ensure that a *valid* data frame is exported. If `data.frame` is completely empty, the content of the text file will be just `""` and the `read.table()` call fails. With a single column, but no rows the data can be exported and imported.